### PR TITLE
feat: ASAP-339 Fetch Leadership data for Team Analytics

### PR DESCRIPTION
**Pull request by peterstarling**

### CODE REVIEW

File: apps/crn-server/test/data-providers/contentful/analytics.data-provider.test.ts [hunk 1]
```diff
@@ -0,0 +1,1365 @@
...
+          );
+
+          const result = await analyticsDataProvider.fetchTeamLeaderShip({});
+
+          expect(result.items[0]!.interestGroupMemberCount).toBe(0);
+        });
+
+        test('Should return interestGroupMemberCount of 2 when the client returns a total of 2 interest groups associated with a team', async () => {
+          const contentfulGraphQLResponse = getAnalyticsTeamLeadershipQuery();
+          contentfulGraphQLResponse.teamsCollection!.items[0]!.linkedFrom!.interestGroupsCollection!.total = 2;
```
Comment (gabiayako) [inline apps/crn-server/test/data-providers/contentful/analytics.data-provider.test.ts:535]: what if the interest group is inactive?

File: apps/crn-server/test/data-providers/contentful/analytics.data-provider.test.ts [hunk 2]
```diff
@@ -0,0 +1,1365 @@
...
+            ];
+          contentfulGraphQLResponse.teamsCollection!.items[0]!.linkedFrom!.teamMembershipCollection!.items[0]!.linkedFrom!.usersCollection!.items.push(
+            {
+              alumniSinceDate: null,
+              linkedFrom: {
+                workingGroupLeadersCollection: {
+                  items: [
+                    {
+                      linkedFrom: {
+                        workingGroupsCollection: {
```
Comment (gabiayako) [inline apps/crn-server/test/data-providers/contentful/analytics.data-provider.test.ts:681]: what if the working group is complete 
Screenshot 2024-03-05 at 08 14 05


or the user is inactive on the working group?
Screenshot 2024-03-05 at 08 14 21

File: apps/crn-server/src/data-providers/contentful/analytics.data-provider.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      ),
+  ) || [];
+const flattenWorkingGroupMember = (user: User): (string | undefined)[] =>
+  user?.linkedFrom?.workingGroupMembersCollection?.items.flatMap(
+    (workingGroupMember) =>
+      workingGroupMember?.linkedFrom?.workingGroupsCollection?.items.map(
+        (item) => item?.sys.id,
+      ),
+  ) || [];
+const getUniqueIdCount = (arr: (string | undefined)[] | undefined): number =>
```
Comment (gabiayako) [inline apps/crn-server/src/data-providers/contentful/analytics.data-provider.ts:140]: I think we need to filter `undefined` first so it doesn't add to the count
Screenshot 2024-03-05 at 07 41 57

**Review by gabiayako — CHANGES_REQUESTED** [review]

File: apps/crn-server/test/data-providers/contentful/analytics.data-provider.test.ts (hunk 1 above)
Comment (peterstarling) [reply apps/crn-server/test/data-providers/contentful/analytics.data-provider.test.ts:535]: good point, might need to raise this with Joao actually

File: apps/crn-server/src/controllers/analytics.controller.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import {
+  FetchPaginationOptions,
+  ListAnalyticsTeamLeadershipResponse,
+} from '@asap-hub/model';
+import { AnalyticsDataProvider } from '../data-providers/types/analytics.data-provider.types';
+
+export default class AnalyticsController {
+  constructor(private analyticsDataProvider: AnalyticsDataProvider) {}
+
+  async fetchTeamLeaderShip(
```
Comment (gabiayako) [inline apps/crn-server/src/controllers/analytics.controller.ts:10]:
```suggestion
  async fetchTeamLeadership(
```

File: apps/crn-server/src/routes/analytics.route.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+): Router => {
+  const analyticsRoutes = Router();
+
+  analyticsRoutes.get(
+    '/team-leadership',
+    async (req, res: Response<ListAnalyticsTeamLeadershipResponse>) => {
+      const parameters = req.query;
+      const query = validateFetchPaginationOptions(parameters);
+
+      const result = await analyticsController.fetchTeamLeaderShip(query);
```
Comment (gabiayako) [inline apps/crn-server/src/routes/analytics.route.ts:17]:
```suggestion
      const result = await analyticsController.fetchTeamLeadership(query);
```

**Review by gabiayako — APPROVED** [review]

**Pull request merged**